modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DocumentLibrary/DocumentLibrary.es.js

```diff
@@ -16,310 +16,280 @@ import '../FieldBase/FieldBase.es';
+				<div className="card-col-content card-col-gutters">
+					<h4 className="text-truncate" title={fileEntryTitle}>
+						{fileEntryTitle}
+					</h4>
+				</div>
+
+				<div className="card-col-field">
+					<a download="{$fileEntryTitle}" href="{$fileEntryURL}">
+						<ClayIcon symbol="download" />
+					</a>
+				</div>
+			</ClayCard.Body>
+		</ClayCard>
+	);
+};
 
-				if (fileEntry.url) {
-					fileEntryURL = fileEntry.url;
-				}
-			}
-			catch (e) {
-				console.warn('Unable to parse JSON', value);
+function transformFileEntryProperties({fileEntryTitle, fileEntryURL, value}) {
+	if (value && typeof value === 'string') {
+		try {
+			const fileEntry = JSON.parse(value);
+
+			fileEntryTitle = fileEntry.title;
+
+			if (fileEntry.url) {
+				fileEntryURL = fileEntry.url;
 			}
 		}
-
-		return {
-			...state,
-			fileEntryTitle,
-			fileEntryURL,
-			value,
-		};
+		catch (e) {
+			console.warn('Unable to parse JSON', value);
+		}
 	}
 
-	getDocumentLibrarySelectorURL() {
-		const {itemSelectorAuthToken} = this.initialConfig_;
-		const {portletNamespace} = this;
-
-		const criterionJSON = {
-			desiredItemSelectorReturnTypes:
-				'com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType,com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType',
-		};
-
-		const uploadCriterionJSON = {
-			URL: this.getUploadURL(),
-			desiredItemSelectorReturnTypes:
-				'com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType,com.liferay.item.selector.criteria.FileEntryItemSelectorReturnType',
-		};
-
-		const documentLibrarySelectorParameters = {
-			'0_json': JSON.stringify(criterionJSON),
-			'1_json': JSON.stringify(criterionJSON),
-			'2_json': JSON.stringify(uploadCriterionJSON),
-			criteria:
-				'com.liferay.item.selector.criteria.file.criterion.FileItemSelectorCriterion',
-			doAsGroupId: themeDisplay.getScopeGroupId(),
-			itemSelectedEventName: `${portletNamespace}selectDocumentLibrary`,
-			p_p_auth: itemSelectorAuthToken,
-			p_p_id: Liferay.PortletKeys.ITEM_SELECTOR,
-			p_p_mode: 'view',
-			p_p_state: 'pop_up',
-			refererGroupId: themeDisplay.getScopeGroupId(),
-		};
-
-		const documentLibrarySelectorURL = createPortletURL(
-			themeDisplay.getLayoutRelativeControlPanelURL(),
-			documentLibrarySelectorParameters
-		);
+	return [fileEntryTitle, fileEntryURL];
+}
 
-		return documentLibrarySelectorURL.toString();
-	}
+const DocumentLibrary = ({
+	fileEntryTitle = '',
+	fileEntryURL = '',
+	id,
+	name,
+	onClearButtonClicked,
+	onSelectButtonClicked,
+	placeholder,
+	readOnly,
+	value,
+}) => {
+	const [transformedFileEntryTitle, transformedFileEntryURL] = useMemo(
+		() =>
+			transformFileEntryProperties({
+				fileEntryTitle,
+				fileEntryURL,
+				value,
+			}),
+		[fileEntryTitle, fileEntryURL, value]
+	);
+
+	return (
+		<div className="liferay-ddm-form-field-document-library">
+			{transformedFileEntryURL && readOnly ? (
+				<CardItem fileEntryTitle={transformedFileEntryTitle} />
```

Comment:
hey @diegonvs I think we can bring `CardItem` here, since we are just using it here.

Reply:
I broke down into another component for a better legibility of the main JSX

Reply:
Yeah, I think it would be nice if this component was too big but apparently it is quite small. So we can put it here.

Reply:
cc @diegonvs